src/google/protobuf/compiler/parser.cc

```diff
@@ -1728,11 +1732,24 @@ bool Parser::ParseReserved(DescriptorProto* message,
   }
 }
 
+bool Parser::ParseReservedName(std::string* name, const char* error_message) {
+  // Capture the position of the token, in case we have to report an
+  // error after it is consumed.
+  int line = input_->current().line;
+  int col = input_->current().column;
+  DO(ConsumeString(name, error_message));
+  if (!io::Tokenizer::IsIdentifier(*name)) {
+    AddWarning(line, col, absl::StrFormat("Reserved name \"%s\" is not a valid identifier.", *name));
+    return false;
```

Comment:
Does this cause the parser to fail?

Reply:
Oops, I bet it does. When I added the `ExpectHasWarnings` test function, looks like I modeled too closely after `ExpectHasErrors`, so it doesn't have the assertion to verify that the operation succeeds 🤦. So even though the test passes, that's likely not enough to verify that the parse succeeded. Fix incoming...

Reply:
Pushed a fix.

Interestingly, it does _not_ cause the parser to fail. It causes the current statement to be aborted/skipped, but the parser keeps going (in the interest of reporting all syntax errors instead of failing fast). But, at the end of the parse, it looks at the number of errors reported to decide if the whole operation was a failure. 

So it was not failing... but it was potentially skipping some of the source and producing an incomplete descriptor.

Even if/when this gets changed back to reporting an error, this function can still return true (since a reserved name *has* been consumed/parsed).

Reply:
Hah good to know!  This LGTM otherwise, lemme rerun the tests and merge it